x/ibc/light-clients/solomachine/types/misbehaviour_handle_test.go

```diff
@@ -12,162 +13,166 @@ func (suite *SoloMachineTestSuite) TestCheckMisbehaviourAndUpdateState() {
+					}
+
+					data, err := suite.chainA.Codec.MarshalBinaryBare(signBytes)
+					suite.Require().NoError(err)
+
+					sig := solomachine.GenerateSignature(data)
+
+					m.SignatureTwo.Signature = sig
+					m.SignatureTwo.Data = msg
+					misbehaviour = m
+				},
+				false,
+			},
+			{
+				"signatures sign over different sequence",
+				func() {
+					clientState = solomachine.ClientState()
 
-				sig, err := suite.solomachine.PrivateKey.Sign(data)
-				suite.Require().NoError(err)
+					// store in temp before assigning to interface type
+					m := solomachine.CreateMisbehaviour()
 
-				m.SignatureTwo.Signature = sig
-				m.SignatureTwo.Data = msg
-				misbehaviour = m
-			},
-			false,
-		},
-		{
-			"signatures sign over different sequence",
-			func() {
-				clientState = suite.solomachine.ClientState()
-
-				// store in temp before assigning to interface type
-				m := suite.solomachine.CreateMisbehaviour()
-
-				// Signature One
-				msg := []byte("DATA ONE")
-				// sequence used is plus 1
-				signBytes := &types.SignBytes{
-					Sequence: suite.solomachine.Sequence + 1,
-					Data:     msg,
-				}
+					// Signature One
+					msg := []byte("DATA ONE")
+					// sequence used is plus 1
+					signBytes := &types.SignBytes{
+						Sequence: solomachine.Sequence + 1,
+						Data:     msg,
+					}
 
-				data, err := suite.chainA.Codec.MarshalBinaryBare(signBytes)
-				suite.Require().NoError(err)
+					data, err := suite.chainA.Codec.MarshalBinaryBare(signBytes)
+					suite.Require().NoError(err)
 
-				sig, err := suite.solomachine.PrivateKey.Sign(data)
-				suite.Require().NoError(err)
+					sig := solomachine.GenerateSignature(data)
 
-				m.SignatureOne.Signature = sig
-				m.SignatureOne.Data = msg
+					m.SignatureOne.Signature = sig
+					m.SignatureOne.Data = msg
 
-				// Signature Two
-				msg = []byte("DATA TWO")
-				// sequence used is minus 1
+					// Signature Two
+					msg = []byte("DATA TWO")
+					// sequence used is minus 1
 
-				signBytes = &types.SignBytes{
-					Sequence: suite.solomachine.Sequence - 1,
-					Data:     msg,
-				}
-				data, err = suite.chainA.Codec.MarshalBinaryBare(signBytes)
-				suite.Require().NoError(err)
+					signBytes = &types.SignBytes{
+						Sequence: solomachine.Sequence - 1,
+						Data:     msg,
+					}
+					data, err = suite.chainA.Codec.MarshalBinaryBare(signBytes)
+					suite.Require().NoError(err)
 
-				sig, err = suite.solomachine.PrivateKey.Sign(data)
-				suite.Require().NoError(err)
+					sig = solomachine.GenerateSignature(data)
 
-				m.SignatureTwo.Signature = sig
-				m.SignatureTwo.Data = msg
+					m.SignatureTwo.Signature = sig
+					m.SignatureTwo.Data = msg
 
-				misbehaviour = m
+					misbehaviour = m
 
+				},
+				false,
 			},
-			false,
-		},
-	}
+		}
 
-	for _, tc := range testCases {
-		tc := tc
+		for _, tc := range testCases {
+			tc := tc
 
-		suite.Run(tc.name, func() {
-			// setup test
-			tc.setup()
+			suite.Run(tc.name, func() {
```

Comment:
If a test fails on the single vs multi for loop, it will output `#01` to show which iteration of the for loop it was on